acceptance: fix allocator test

tbg opened the pull request:
Currently running this 6x to see that it passes (or at least gets past the interesting point).

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/7625)

<!-- Reviewable:end -->

tbg committed:
acceptance: clean up after alloc test, update docs

tamird commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[acceptance/allocator_test.go, line 42 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7625#-KLwaps50xwHDo_oaRdS:-KLwaps50xwHDo_oaRdT:1783496095) ([raw file](https://github.com/cockroachdb/cockroach/blob/c1a45180aee8f58056e24d98b2a19b27fedb3975/acceptance/allocator_test.go#L42)):*
> ```Go
> // relative to `./acceptance/allocator_terraform` and that the binary is
> // a linux binary (i.e. built via `./build/builder.sh make build` unless
> // when running Linux locally):
> ```

remove this unless clause -- see #7477 

---

*[acceptance/allocator_test.go, line 122 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7625#-KLwb7edM79ahsFnq4Bj:-KLwb7eeyUFpNCdH1OmN:1582446541) ([raw file](https://github.com/cockroachdb/cockroach/blob/c1a45180aee8f58056e24d98b2a19b27fedb3975/acceptance/allocator_test.go#L122)):*
> ```Go
> 		baseDir := filepath.Join(wd, at.f.Cwd)
> 		if t.Failed() && at.f.KeepClusterAfterFail {
> 			t.Logf("test has failed, not destroying; run:\n(cd %s && terraform destroy -state %s)",
> ```

nit: `t.Logf` here but `log.Warning` below?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7625#-:-KLwbHZgLW_TBmbsiDGM:1417088721)*
<!-- Sent from Reviewable.io -->

cuongdo commented on acceptance/allocator_test.go:
```diff
@@ -155,6 +150,16 @@ func (at *allocatorTest) Run(t *testing.T) {
 	at.f.Assert(t)
 	log.Info("initial cluster is up")
 
+	// We must stop the cluster because a) `nodetool` pokes at the data directory
```
s/nodetool/nodectl/

cuongdo commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7625#-:-KLwqkBAWL9LrmQKTa3J:1417088721)*
<!-- Sent from Reviewable.io -->

tbg committed:
acceptance: fix allocator tests

The tests were sloppy in setting up the initial cluster with
the downloaded data directories. There was a window of time
during which the pre-restore cluster and the post-restore
cluster had nodes running simultaneously. See #7224.
